source/common/upstream/eds.cc

```diff
@@ -105,6 +108,17 @@ void EdsClusterImpl::onConfigUpdate(const ResourceVector& resources) {
                                  locality_weights_map_[i], priority_state[i].second);
     }
   }
+
+  // Loop over all priorities not present in the config that already exists. This will
+  // empty out any remaining priority that the config update did not refer to.
+  for (size_t i = priority_state.size(); i < priority_set_.hostSetsPerPriority().size(); ++i) {
+    const HostVector empty_hosts;
+    LocalityWeightsMap empty_locality_map;
+
+    cluster_rebuilt |= updateHostsPerLocality(priority_set_.getOrCreateHostSet(i), empty_hosts,
```

Comment:
Unclear to me whether we also want to delete the `HostSet` for priorities that are emptied out. Wanted to get input from reviewers before putting in the work to do so as it makes the solution much less trivial. The current solution will leave an empty `HostSet` around forever, but I don't imagine the cost of that is super high?

Reply:
If this is simpler I think it's fine to leave it until anyone needs that particular optimization.    Honestly I'd think we'd want to delete the entire priority level and that's definitely non-trivial!